src/login.ts

```diff
@@ -1,57 +1,95 @@
-};
+export type RequestParams = Pick<
+  MastoConfigProps,
+  | 'url'
+  | 'logLevel'
+  | 'timeout'
+  | 'defaultRequestInit'
+  | 'disableDeprecatedWarning'
+>;
 
-export const login = async (params: LoginParams): Promise<Client> => {
-  const draft: Writable<MastoConfigProps> = {
+const buildHttpContext = (
+  params: RequestParams,
+): {
+  serializer: SerializerNativeImpl;
+  config: MastoConfig;
+  logger: Logger;
+  http: HttpNativeImpl;
+} => {
+  const props = {
     url: params.url,
     streamingApiUrl: '',
     logLevel: params.logLevel,
-    accessToken: params.accessToken,
     timeout: params.timeout,
     defaultRequestInit: params.defaultRequestInit,
-    disableVersionCheck: params.disableVersionCheck,
     disableDeprecatedWarning: params.disableDeprecatedWarning,
   };
   const serializer = new SerializerNativeImpl();
+  const config = new MastoConfig(props, serializer);
+  const logger = new LoggerConsoleImpl(config.getLogLevel());
+  const http = new HttpNativeImpl(serializer, config, logger);
+  return { serializer, config, logger, http };
+};
 
-  {
-    const config = new MastoConfig(draft, serializer);
-    const http = new HttpNativeImpl(serializer, config);
-    const instance = await new InstanceRepository(http, config).fetch();
-    draft.version = new SemVer(instance.version);
-    draft.streamingApiUrl = instance.urls.streamingApi;
-  }
+export const fetchV1Instance = (
+  params: RequestParams,
+): Promise<v1.Instance> => {
+  const { http, config } = buildHttpContext(params);
+  return new V1InstanceRepository(http, config).fetch();
+};
 
-  const config = new MastoConfig(draft, serializer);
-  const logger = new LoggerConsoleImpl(config.getLogLevel());
+export const fetchV2Instance = (
+  params: RequestParams,
+): Promise<v2.Instance> => {
+  const { http, config } = buildHttpContext(params);
+  return new V2InstanceRepository(http, config).fetch();
+};
+
+export type CreateClientParams = Omit<MastoConfigProps, 'version'> & {
+  readonly version?: string;
+};
+
+export const createClient = (params: CreateClientParams): Client => {
+  const props: Writable<MastoConfigProps> = {
+    ...params,
+    version: params.version ? new SemVer(params.version) : undefined,
+  };
+  const { serializer, config, logger, http } = buildHttpContext(props);
   const ws = new WsNativeImpl(config, serializer);
-  const http = new HttpNativeImpl(serializer, config, logger);
 
   logger.debug('Masto.js initialised', config);
   return new Client(http, ws, config);
 };
+
+export type LoginParams = Omit<
+  CreateClientParams,
+  'streamingApiUrl' | 'version'
+>;
+
+/**
+ * Fetching instance information and create a client
+ *
+ * Shortcut of `fetchV1Instance` and `createClient`
+ */
+export const login = async (params: LoginParams): Promise<Client> => {
+  const instance = await fetchV1Instance(params);
+  return createClient({
+    ...params,
+    version: instance.version,
+    streamingApiUrl: instance.urls.streamingApi,
```

Comment:
If you specify `accessToken` or something unique to `CreateClientParams` and `LoginParams` it will be ignored because it will be passed to `buildHttpContext` that will only pick properties in `RequestParams`.

<img width="865" alt="image" src="https://user-images.githubusercontent.com/19276905/211814662-e4152597-cde6-4256-b37e-f73bb0d15784.png">

Reply:
I'm sorry for the lack of test codes, but it appears to be still happening. The problem is in these lines 🙇 
https://github.com/sxzz/masto.js/blob/6991bf1c2a3ace7bb9139bf6ca8eeeae2f56f86e/src/login.ts#L33-L40

Reply:
done.